.github/workflows/test-linux-distros.yaml

```diff
@@ -0,0 +1,250 @@
+                xz-devel libuuid-devel
+            elif [[ "$DISTRO" == *"centos:stream8"* ]]; then
+              # CentOS Stream 8 specific handling
+              if ls /etc/yum.repos.d/CentOS-*.repo &>/dev/null; then
+                sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-*.repo
+                sed -i 's|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-*.repo
+              fi
+              
+              # Enable repositories directly
+              dnf -y install dnf-plugins-core
+              dnf -y config-manager --set-enabled powertools || true
+              dnf -y install epel-release || dnf -y install https://dl.fedoraproject.org/pub/epel/epel-release-latest-$(rpm -E %rhel).noarch.rpm
+              dnf -y install git curl make gcc zlib-devel openssl-devel bzip2 \
+                bzip2-devel readline-devel sqlite-devel tk-devel libffi-devel \
+                xz-devel libuuid-devel
+            elif command -v apt-get &> /dev/null; then
+              export DEBIAN_FRONTEND=noninteractive
+              apt-get update
+              apt-get install -y git curl make gcc zlib1g-dev \
+                libssl-dev libbz2-dev libreadline-dev libsqlite3-dev tk-dev \
+                libffi-dev liblzma-dev uuid-dev
+            elif command -v dnf &> /dev/null; then
+              dnf install -y dnf-plugins-core
+              dnf config-manager --set-enabled crb || dnf config-manager --set-enabled powertools || true
+              dnf install -y --allowerasing git curl make gcc zlib-devel openssl-devel bzip2 \
+                bzip2-devel readline-devel sqlite-devel tk-devel libffi-devel \
+                xz-devel libuuid-devel
+            fi
+          }
+
+          # Run the setup
+          setup_environment
+          SETUPEOF
+
+          chmod +x setup.sh
+
+          # Now create the Dockerfile that uses this script
+          cat > Dockerfile.test << EOF
+          FROM ${{ matrix.distro.image }}
+
+          # Copy setup script
+          COPY setup.sh /setup.sh
+
+          # Set the DISTRO environment variable and run setup
+          ENV DISTRO="${{ matrix.distro.image }}"
+          RUN /setup.sh
+
+          # Install pyenv
+          ENV PYENV_ROOT="/root/.pyenv"
+          ENV PATH="\$PYENV_ROOT/bin:\$PYENV_ROOT/shims:\$PATH"
+
+          RUN git clone https://github.com/pyenv/pyenv.git \$PYENV_ROOT && \
+              cd \$PYENV_ROOT && git checkout master
+
+          # Install Python 3.12 using pyenv
+          RUN PYTHON_VERSION="3.12.9" && \
+              CONFIGURE_OPTS="--enable-shared" pyenv install \$PYTHON_VERSION && \
+              pyenv global \$PYTHON_VERSION
+
+          # Install poetry
+          RUN pip install poetry
+
+          # Create workspace directory
+          RUN mkdir -p /workspace
+          WORKDIR /workspace
+          EOF
+
+      - name: Build and push Docker image
```

Comment:
Why do we need to publish images to DockerHub to test linux binaries?

Reply:
@achidlow its not pushing data to dockerhub though. Images are pushed to the repos ghcr and are only used for caching to speed up CI instead of building the images from scratch on each load.

Reply:
I think i got what you mean though, I'll see if it's possible to remove the push flag while still using cache-to - which should cache the intermediate layers only without pushing the finale image.

Reply:
Yeah nice, I saw how much it speeds things up when I tried to remove it lol

Reply:
I've managed to remove the need for docker at all when testing in my branch, will PR into this one once I'm done